keras_hub/src/utils/openvino_utils.py

```diff
@@ -0,0 +1,311 @@
+def setup_openvino_test_config(config_file_path):
+    """Setup OpenVINO test configuration with whitelist approach.
+
+    Args:
+        config_file_path: Path to the config file directory.
+
+    Returns:
+        list: Supported paths (whitelist) for OpenVINO testing.
+    """
+    return load_openvino_supported_tools(
+        Path(config_file_path) / "openvino_supported_tests.txt"
+    )
+
+
+@functools.lru_cache(maxsize=256)
+def _contains_training_methods(file_path, test_name):
+    """Check if a test function contains training methods.
+
+    Args:
+        file_path: Path to the test file.
+        test_name: Name of the test function.
+
+    Returns:
+        bool: True if training methods found, False otherwise.
+    """
+    training_methods = {
+        "fit",
+        "fit_generator",
+        "train_on_batch",
+        "compile",
+        "train_step",
+        "train",
+        "backward",
+        "zero_grad",
+        "step",
+    }
+
+    training_keywords = {
+        "optimizer",
+        "loss",
+        "epochs",
+        "batch_size",
+        "learning_rate",
+    }
+
+    training_test_methods = {
+        "run_layer_test",
+        "run_training_step",
+        "run_build_asserts",
+        "run_task_test",
+        "run_preprocessing_layer_test",
+    }
+
+    class TrainingMethodDetector(ast.NodeVisitor):
+        def __init__(self):
+            self.has_training_methods = False
+
+        def visit_Call(self, node):
+            if (
+                hasattr(node.func, "attr")
+                and node.func.attr in training_methods
+            ):
+                self.has_training_methods = True
+
+            if (
+                hasattr(node.func, "attr")
+                and node.func.attr in training_test_methods
+            ):
+                self.has_training_methods = True
+
+            if (
+                hasattr(node.func, "value")
+                and hasattr(node.func.value, "id")
+                and node.func.value.id == "self"
+                and hasattr(node.func, "attr")
+                and node.func.attr in training_test_methods
+            ):
+                self.has_training_methods = True
+
+            self.generic_visit(node)
+
+        def visit_keyword(self, node):
+            """Visit keyword arguments to detect training keywords."""
+            if node.arg in training_keywords:
+                self.has_training_methods = True
+            self.generic_visit(node)
+
+    try:
+        with open(file_path, "r", encoding="utf-8") as f:
+            source = f.read()
+        tree = ast.parse(source)
+        for node in ast.walk(tree):
+            if isinstance(node, ast.FunctionDef) and node.name == test_name:
+                detector = TrainingMethodDetector()
+                detector.visit(node)
+                return detector.has_training_methods
+        return False
+    except (OSError, SyntaxError):
+        return True
+
+
+def should_auto_skip_training_test(item):
```

Comment:
let's try to move all this test related stuff out of here. Ideally the whole test configuring for open vino is ~20-30 lines of code straight in conftest.py. If we need more (what we currently have still feels overwrought to me), let's move it to a open vino specific file in `tests/`.

But we should keep the code needed at runtime for the library (compile stuff below), that we ship with the library, separate from any test only stuff, which we will not.